Sources/Brave/WebFilters/ContentBlocker/ContentBlockerManager.swift

```diff
@@ -313,53 +313,44 @@ final public class ContentBlockerManager: Sendable {
     return ruleList
   }
   
-  /// Return the enabled rule types for this domain and the enabled settings
-  @MainActor public func compiledRuleTypes(for domain: Domain) -> Set<RuleTypeWithSourceType> {
+  @MainActor public func enabledRuleTypes(for domain: Domain) -> Set<ContentBlockerManager.BlocklistRuleType> {
     let filterLists = FilterListResourceDownloader.shared.filterLists
     
     if domain.shield_allOff == 1 {
       return []
     }
     
-    var results = Set<RuleTypeWithSourceType>()
+    var results = Set<ContentBlockerManager.BlocklistRuleType>()
 
     // Get domain specific rule types
     if domain.isShieldExpected(.AdblockAndTp, considerAllShieldsOption: true) {
-      if let sourceType = self.sourceType(for: .general(.blockAds)) {
-        results.insert(RuleTypeWithSourceType(ruleType: .general(.blockAds), sourceType: sourceType))
-      }
-      
-      if let sourceType = self.sourceType(for: .general(.blockTrackers)) {
-        results.insert(RuleTypeWithSourceType(ruleType: .general(.blockTrackers), sourceType: sourceType))
-      }
+      results.insert(.general(.blockAds))
+      results.insert(.general(.blockTrackers))
     }
     
     // Get filter list specific rule types
     filterLists.forEach { filterList in
       guard filterList.isEnabled else { return }
-      let ruleType = filterList.makeRuleType()
-      
-      guard let sourceType = self.sourceType(for: ruleType) else {
-        return
-      }
-      
-      guard cachedRuleList(for: ruleType) != nil else {
-        return
-      }
-      
-      results.insert(RuleTypeWithSourceType(ruleType: ruleType, sourceType: sourceType))
+      results.insert(filterList.makeRuleType())
     }
     
     // Get global rule types
     if Preferences.Privacy.blockAllCookies.value {
-      if let sourceType = sourceType(for: .general(.blockCookies)) {
-        results.insert(RuleTypeWithSourceType(ruleType: .general(.blockCookies), sourceType: sourceType))
-      }
+      results.insert(.general(.blockCookies))
     }
     
     return results
   }
   
+  /// Return the enabled rule types for this domain and the enabled settings
+  @MainActor public func ruleLists(for domain: Domain) async -> Set<WKContentRuleList> {
+    let ruleTypes = enabledRuleTypes(for: domain)
+    
+    return await Set(ruleTypes.asyncConcurrentCompactMap { ruleType in
```

Comment:
`asyncConcurrentCompactMap` also not needed here, `cachedRuleList` is not async

Reply:
Removed the async. It was async because of the cherry pick which had some internals async. Thought this contentBlockerManager is going to completely change.